python/heterocl/rptdisp.py

```diff
@@ -0,0 +1,338 @@
+import pandas as pd
+from tabulate import tabulate
+
+class RptDisp(object):
+  """
+  Queryable report displayer.
+
+  ...
+
+  Attributes
+  ----------
+  _category: list
+      List of default latency information category.
+
+  _category_aux: list
+      List of latency information category with range indicators.
+
+  _loop_name: list
+      List of loop names without loop nest indicators.
+
+  _loop_name_aux: list
+      List of loop names with loop nest indicators.
+
+  _max_level: int
+      Maximum level of loop nest in the report file.
+
+  _data: dict
+      Dictionary containing latency data. 
+
+  unit: str
+      Unit of each information.
+
+  Methods
+  ----------
+  __is_range(v, key)
+      Check whether the value is a range value or not.
+
+  scan_range(obj)
+      Scan the entire report file to see which latency category contains
+      range values.
+
+  __get_value(v, key, minmax)
+      Get the value associated with the input key.
+
+  __info_extract(obj, key, minmax, col)
+      Extract out all the latency information from the report. 
+
+  init_data(obj)
+      Initialize the data given the report file.  
+
+  get_loops(obj)
+      Acquire loop names with and without loop nest indicators.
+  
+  __select_loops(loops)
+      Select only the loops specified by the user.
+  
+  __select_levels(loops, level)
+      Select only the loops that are included in the level specified by
+      the user.
+  
+  __select_cols(cols)
+      Select only the columns specified by the user.
+  
+  display(loops=None, level=None, cols=None)
+      Display the report table with appropriate query arguments.
+  """
+
+  def __init__(self, unit):
+    """
+    Parameters
+    ----------
+    unit: str
+        Unit for all numerical values in the table.
+    """
+    self._category = ['Trip Count', 'Latency', 'Iteration Latency', 
+                       'Pipeline II', 'Pipeline Depth']
+    self._category_aux = []
+    self._loop_name = []
+    self._loop_name_aux = []
+    self._max_level = 0
+    self._data = {}
+    self.unit = unit
+
+  def __is_range(self, v, key):    
+    """Check whether the value is a range value or not.
+
+    Parameters
+    ----------
+    v: dict
+        Dictionary containing all latency information for a particular loop.
+    key: str
+        Latency category.
+
+    Returns
+    ----------
+    bool
+      True if the value is a range value. Otherwise, False. 
+    """
+    return isinstance(v.get(key), dict)
+
+  def scan_range(self, obj):
+    """Scans the parsed xml file to check which categories have range 
+    values and updates _category_aux accordingly. Also, it initializes
+    _data to be used in displaying the report data.
+
+    Parameters
+    ----------
+    obj: dict 
+        Dictionary representation of the report file.
+
+    Returns
+    ----------
+    None
+    """
+    detect_minmax = []
+    for item in self._category:
+      cat = item.replace(' ', '')
+      tf = False
```

Comment:
Try making variable names more meaningful. I couldn't interpret what `tf` stands for. Also, a golden rule is to avoid using variable names <= two characters.